From 50f99101cd418a954bea7198e3a5a4bf1770c6e4 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Fri, 10 Feb 2017 11:03:30 -0600
Subject: [PATCH 1504/5242] MLK-13906: ARM: dts: new dtb for mfgtool to burn
 both BSP and M4 to eMMC

commit  0c2cc7d1912e9ab1bdb3bad688096a022f06bdd2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add one more dtb file for mfgtool purpose only, it enabled both eMMC and
QSPI to burn both BSP and M4 images in one process.

Kernel still uses the original eMMC dtb so the QSPI, which belongs to
M4 domain won't be exposed in A7 domain.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                  |    1 +
 arch/arm/boot/dts/imx7ulp-evk-emmc-qspi.dts |   18 ++++++++++++++++++
 2 files changed, 19 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx7ulp-evk-emmc-qspi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 77749d4..4d3b2b7 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -698,6 +698,7 @@ dtb-$(CONFIG_SOC_IMX7ULP) += \
 	imx7ulp-14x14-arm2.dtb \
 	imx7ulp-evk.dtb \
 	imx7ulp-evk-emmc.dtb \
+	imx7ulp-evk-emmc-qspi.dtb \
 	imx7ulp-evk-sd1.dtb \
 	imx7ulp-evk-lpuart.dtb \
 	imx7ulp-evk-qspi.dtb \
diff --git a/arch/arm/boot/dts/imx7ulp-evk-emmc-qspi.dts b/arch/arm/boot/dts/imx7ulp-evk-emmc-qspi.dts
new file mode 100644
index 0000000..47289c6
--- /dev/null
+++ b/arch/arm/boot/dts/imx7ulp-evk-emmc-qspi.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7ulp-evk-qspi.dts"
+
+&usdhc0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&pinctrl_usdhc0_8bit>;
+	pinctrl-1 = <&pinctrl_usdhc0_8bit>;
+	non-removable;
+	bus-width = <8>;
+	status = "okay";
+};
-- 
1.7.9.5

